dbt/adapters/bigquery/relation.py

```diff
@@ -53,6 +65,42 @@ def project(self):
     def dataset(self):
         return self.schema
 
+    @classmethod
+    def materialized_view_from_model_node(
+        cls, model_node: ModelNode
+    ) -> BigQueryMaterializedViewConfig:
+        return BigQueryMaterializedViewConfig.from_model_node(model_node)  # type: ignore
+
+    @classmethod
+    def materialized_view_config_changeset(
+        cls,
+        existing_materialized_view: BigQueryMaterializedViewConfig,
+        runtime_config: RuntimeConfigObject,
+    ) -> Optional[BigQueryMaterializedViewConfigChangeset]:
+        config_change_collection = BigQueryMaterializedViewConfigChangeset()
+        new_materialized_view = cls.materialized_view_from_model_node(runtime_config.model)
+        assert isinstance(existing_materialized_view, BigQueryMaterializedViewConfig)
+        assert isinstance(new_materialized_view, BigQueryMaterializedViewConfig)
+
+        if new_materialized_view.options != existing_materialized_view.options:
+            config_change_collection.options = BigQueryOptionsConfigChange(
+                action=RelationConfigChangeAction.alter,
+                context=new_materialized_view.options,
+            )
+
+        if (
+            new_materialized_view.cluster != existing_materialized_view.cluster
```

Comment:
What about a missing cluster on the existing materialized view?

Reply:
This was done because the type on context is not optional. But None is a valid value if the new object does not have clustering. I updated the type instead.